tests/Doctrine/Tests/ORM/Functional/Ticket/GH8499Test.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use DateTime;
+use DateTimeInterface;
+use Doctrine\DBAL\LockMode;
+use Doctrine\DBAL\Schema\SchemaException;
+use Doctrine\ORM\Mapping\Column;
+use Doctrine\ORM\Mapping\Entity;
+use Doctrine\ORM\Mapping\GeneratedValue;
+use Doctrine\ORM\Mapping\Id;
+use Doctrine\ORM\Mapping\Table;
+use Doctrine\ORM\Mapping\Version;
+use Doctrine\ORM\OptimisticLockException;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+use function sleep;
+
+class GH8499Test extends OrmFunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        try {
+            $this->setUpEntitySchema(
+                [GH8499VersionableEntity::class]
+            );
+        } catch (SchemaException $e) {
+        }
+    }
+
+    /**
+     * @group GH-8499
+     */
+    public function testOptimisticLockInitializesVersionOnEntityInsert(): void
+    {
+        $entity = new GH8499VersionableEntity();
+        $entity->setName('Test Entity');
+        $entity->setDescription('Entity to test optimistic lock fix with DateTimeInterface objects');
+        $this->_em->persist($entity);
+        $this->_em->flush();
+        $this->_em->refresh($entity);
+
+        $this->assertInstanceOf(DateTimeInterface::class, $entity->getRevision(), 'Version field not set to DateTimeInterface');
+    }
+
+    /**
+     * @group GH-8499
+     */
+    public function testOptimisticLockUpdatesVersionOnEntityUpdate(): void
+    {
+        $entity = new GH8499VersionableEntity();
+        $entity->setName('Test Entity');
+        $entity->setDescription('Entity to test optimistic lock fix with DateTimeInterface objects');
+        $this->_em->persist($entity);
+        $this->_em->flush();
+        $this->_em->refresh($entity);
+        $firstVersion = $entity->getRevision();
+        sleep(1);
```

Comment:
I wish there was a better way to do this, maybe another maintainer will have a better idea. Right now, I think this test is probably the longest test of the whole test suite.

Reply:
I have given it more thought:
That feature also works with integer Versions.
You already tested that the version was initialized on insert. If an entity's integer version is updated on update, I think we can reasonably think it will work too for a datetime version.

1. Are we sure such a test does not already exist?
2. Should we rewrite that particular test with an integer field so that we can avoid that call to `sleep`? And if yes, maybe it shouldn't be in `GH9499Test`. This test passes without your fix, doesn't it?

Reply:
Use dbal to update the row into the past